packages/supermassive/src/executeWithoutSchema.ts

```diff
@@ -1921,15 +2075,28 @@ function invokeBeforeOperationExecuteHook(exeContext: ExecutionContext) {
         context: exeContext.contextValue,
         operation: exeContext.operation,
       }),
-    (_, rawError) => {
+    (result, rawError) => {
       if (rawError) {
         const error = toGraphQLError(
           rawError,
           undefined,
           "Unexpected error in beforeOperationExecute hook",
         );
         exeContext.errors.push(error);
+
+        return error;
       }
+
+      if (result instanceof Error) {
+        const error = toGraphQLError(
+          result,
+          undefined,
+          "Unexpected error in beforeOperationExecute hook",
```

Comment:
Let's:

1. Differentiate those two cases: the returned value and thrown
2. Include operation name in the message (if it is set)?

```suggestion
          `Unexpected error returned from beforeOperationExecute hook (operation: ${operationName})`,
```